src/atopile/front_end.py

```diff
@@ -323,13 +331,15 @@ def visitImport_stmt(
                 )
                 for name_or_attr in ctx.name_or_attr()
             ]
-            return KeyOptMap(KeyOptItem.from_kv(li.ref, li) for li in lazy_imports)
+            return KeyOptMap(
+                KeyOptItem.from_kv(li.ref, (li, ctx)) for li in lazy_imports
+            )
 
         else:
             # Standard library imports are special, and don't require a from path
             imports = []
             for collector, name_or_attr in iter_through_errors(ctx.name_or_attr()):
-                with collector():
+                with collector(), log_user_errors():
```

Comment:
This change means that we end up with the errors + warnings properly in sequence and interlaced w/ any other build info